### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by freeformz**
This is on top of the data_race branch.

### CODE REVIEW

File: http_outlet.go [hunk 1]
```diff
@@ -136,13 +136,13 @@ func (h *HTTPOutlet) retryPost(batch Batch) {
...
 
-func (h *HTTPOutlet) post(formatter HTTPFormatter, uuid string) error {
+func (h *HTTPOutlet) post(formatter HTTPFormatter) error {
 	req, err := formatter.Request()
 	if err != nil {
 		return err
 	}
 
-	req.Header.Add("X-Request-Id", uuid)
+	uuid := req.Header.Get("X-Request-Id")
```
Comment (apg) [inline http_outlet.go]: This is a bit leaky. I'd rather see HTTPFormatter gain the ability to return a RequestID, even if it's stored in the header down below.

Comment (apg) [inline http_outlet.go]: Though, the Kinesis formatter probably need not know of a request id... hrrmm.

Comment (freeformz) [inline http_outlet.go]: The only reason we extract it here is that it's used in some logging below. It's part of the request, which the formatter generates from the batch, so this seems like the right place / way to get it.

Comment (apg) [inline http_outlet.go]: :ok: then.

**Pull request merged**